src/client/Microsoft.Identity.Client/Cache/Prototype/IdentityCacheWrapper.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.Identity.ServiceEssentials;
+
+namespace Microsoft.Identity.Client.Cache.Prototype
+{
+    internal class IdentityCacheWrapper
+    {
+        internal static IIdentityCache s_iIdentityCache { get; set; }
+
+        internal IdentityCacheWrapper(CacheOptions cacheOptions)
+        {
+            // Set (or overwrite) cache to user-specified implementation, otherwise set to default implementation, if not already set.
```

Comment:
not sure about default implentation.

Reply:
Yea, definitely it's adding a dependency to .NET Memory Cache. Added it for simplicity to show that it could work E2E. Practically we could replace it with some other memory cache with eviction that doesn't have dependencies (eg MiseCache).